compiler/rustc_hir_analysis/src/collect.rs

```diff
@@ -444,6 +449,96 @@ impl<'tcx> HirTyLowerer<'tcx> for ItemCtxt<'tcx> {
         self.tcx.at(span).type_param_predicates((self.item_def_id, def_id, assoc_ident))
     }
 
+    fn select_inherent_assoc_candidates(
+        &self,
+        span: Span,
+        self_ty: Ty<'tcx>,
+        candidates: &Vec<InherentAssocCandidate>,
+    ) -> (Vec<InherentAssocCandidate>, Vec<FulfillmentError<'tcx>>) {
+        // This is all rather hacky. We attempt to unify impl headers against this self ty
+        // (even though we aren't really in a position to do so) as users of IATs in signatures
+        // are unable to explicitly disambiguate which impl to use other than by specifying the
+        // self type.
+
+        struct ReplaceAliasesWithInfer<'tcx, 'a>(&'a InferCtxt<'tcx>);
```

Comment:
Do we need this folder w/ the new solver? We could just relate and throw away all the alias-relate predicates?

Reply:
Yeah with the new solver this is unnecessary